lib/util/tXml.js

```diff
@@ -815,6 +821,27 @@ shaka.util.TXml = class {
   }
 
 
+  /**
+   * Search the node index by the t attribute
+   * and return the index. if not found return null
+   * @param {!Array<shaka.extern.xml.Node>} nodes
+   * @param {!number} t
+   * @private
+   */
+  static nodePositionByAttributeT_(nodes, t) {
```

Comment:
Can you make this method to work on other fields via additional parameter? It would be great to reuse it for an `n` attribute later

Reply:
Yes sure

Reply:
done

Reply:
@Iragne do you want implement the support to `n` also in this PR? Thanks!

Reply:
@avelad happy to do it too. sadly I have no stream to test the @n= but i will do it.
Additionally I would like to add mor ein this PR because i also correct the adaptation set indexing and not by id.

mainly a patch manifest with 
    <add sel="/MPD/Period[@id=&#39;424242&#39;]/AdaptationSet[2]/SegmentTemplate/SegmentTimeline"><S t="120123334" d="40106666"></S></add>

Reply:
And please, update the PR title!

Reply:
done